mongo/datadog_checks/mongo/api.py

```diff
@@ -9,36 +8,12 @@
 
 
 class MongoApi(object):
-    def __init__(self, config, log, replicaset=None):
+    # :params replicaset: If replication is enabled, this parameter specifies the name of the replicaset.
+    # Valid for ReplicaSetDeployment deployments
+    def __init__(self, config, log, replicaset: str = None):
         self._config = config
         self._log = log
-        self.deployment_type = None
-        if self._config.server:
-            # Deprecated option
-            self._cli = MongoClient(
-                self._config.server,
-                socketTimeoutMS=self._config.timeout,
-                connectTimeoutMS=self._config.timeout,
-                serverSelectionTimeoutMS=self._config.timeout,
-                read_preference=ReadPreference.PRIMARY_PREFERRED,
-                replicaSet=replicaset,
-                **self._config.ssl_params
-            )
-        else:
-            merged_options_dict = {}
-            merged_options_dict.update(self._config.additional_options)
-            merged_options_dict.update(self._config.ssl_params)
-            self._cli = MongoClient(
-                host=self._config.hosts,
-                socketTimeoutMS=self._config.timeout,
-                connectTimeoutMS=self._config.timeout,
-                serverSelectionTimeoutMS=self._config.timeout,
-                read_preference=ReadPreference.PRIMARY_PREFERRED,
-                replicaSet=replicaset,
-                appname=DD_APP_NAME,
-                **merged_options_dict
-            )
-        self._initialize()
+        self._initialize(replicaset)
```

Comment:
feels redundant to have a method for this

Reply:
Done